modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -22,17 +22,18 @@
 import pandas
 from pandas.core.indexes.api import ensure_index, Index, RangeIndex
 from pandas.core.dtypes.common import is_numeric_dtype, is_list_like
-from typing import List, Hashable
+from typing import List, Hashable, Optional, Callable, Union, Dict
 
 from modin.core.storage_formats.pandas.query_compiler import PandasQueryCompiler
 from modin.error_message import ErrorMessage
 from modin.core.storage_formats.pandas.parsers import (
     find_common_type_cat as find_common_type,
 )
+from modin.core.dataframe.base.dataframe.dataframe import ModinDataframe
 from modin.pandas.indexing import is_range_like
 
 
-class PandasDataframe(object):
+class PandasDataframe(ModinDataframe):
```

Comment:
I think this class could use a rename. @YarShev @vnlitvinov @dchigarev thoughts? What about splitting the pandas memory format specifics into a different class and mix it in? That way we could reuse this logic for other memory formats (e.g. Pyarrow)

```suggestion
class PartitionedDataframe(ModinDataframe):
```

Reply:
That makes sense if there's a clean cut... someone should try doing it, probably.

Reply:
I agree that makes sense to rename this class to reuse the logic for other memory formats. However, won't `PartitionedDataframe` be confusing with respect to base `ModinDataframe`? We pose a Modin Dataframe as a partitioned dataframe by nature. Maybe we should name base class as `ModinDataframeIface` and child partitioned dataframe that is common for storage formats as `ModinDataframe`. What do you think?

Reply:
cc @devin-petersohn, @RehanSD, @vnlitvinov

Reply:
Based off of our meeting today (12/17) we came up with the following two proposals:
* Igor: We could name the parent class (`ModinDataframe`) the `BaseModinDataframe` (to match our naming scheme for the query compiler), and rename this file (`PandasDataframe`) to `ModinDatafame`.
* Rehan: We could keep the name for `ModinDataframe`, since it just exposes the algebra, and then rename the `PandasDataframe` the `PartitionedModinDataframe` since it implements the algebra but also knows that its partitioned.

Both solutions will involve separating out the `pandas` specific logic into a mixin.

Reply:
@Garra1980

Reply:
Possible, `Base*` prefix isn't correct for this case. `Base*` class assumes some basic implementation of functionality, but in the current `ModinDataframe` we have only abstract methods. This abstractness should be highlighted in the class name, for example `ABCModinDataframe` - pandas also uses the approach with `ABC` prefix for generic base entities naming (`ABCSeries`, `ABCIndex` e.g.). In this case we naming will be `ModinDataframe`->`ABCModinDataframe`, `PandasDataframe`->`ModinDataframe`

Reply:
I think this suggestion makes the most sense to me:

>We could keep the name for ModinDataframe, since it just exposes the algebra, and then rename the PandasDataframe the PartitionedModinDataframe since it implements the algebra but also knows that its partitioned.

Reply:
+1
> We could keep the name for ModinDataframe, since it just exposes the algebra, and then rename the PandasDataframe the PartitionedModinDataframe since it implements the algebra but also knows that its partitioned.

Reply:
Aside note: I wonder how we could name dataframe classes specific for storage formats - PartitionedModinPandasDataframe, PartitionedModinPyarrowDataframe, etc.

Reply:
Renaming `PandasDataframe` should probably go in a separate PR.